halo2_gadgets/src/poseidon/primitives/mds.rs

```diff
@@ -82,7 +82,12 @@ pub(super) fn generate_mds<F: FromUniformBytes<64> + Ord, const T: usize>(
                 acc
             } else {
                 // We can invert freely; by construction, the elements of xs are distinct.
-                acc * (x - x_m) * (x_j - x_m).invert().unwrap()
+                
+                // We hard-code the type, to avoid wasm-pack compiler error of: cannot infer type. 
+                let denominator: F = x_j - x_m;
+                let denominator_inverted: F = denominator.invert().unwrap();
+                
+                acc * (x - x_m) * denominator_inverted
```

Comment:
```suggestion
                // We hard-code the type, to avoid wasm-pack compiler error of: cannot infer type. 
                let denominator: F = x_j - x_m;

                // We can invert freely; by construction, the elements of xs are distinct.
                let denominator_inverted: F = denominator.invert().unwrap();

                acc * (x - x_m) * denominator_inverted
```